src/test/run-make/codegen-options-parsing/Makefile

```diff
@@ -22,3 +22,10 @@ all:
 	$(RUSTC) -C lto=foo dummy.rs 2>&1 | \
 		grep 'codegen option `lto` takes no value'
 	$(RUSTC) -C lto dummy.rs
+
+	# Should contain gc-sections...
+	$(RUSTC) -Z print-link-args dummy.rs 2>&1 | \
+	  grep '"-Wl,--gc-sections"'
```

Comment:
I discussed with @pnkfelix [about testing](https://github.com/rust-lang/rust/pull/31368#issuecomment-179221480) also on IRC. A better approach would be to use
```makefile
objdump --disassemble $(TMPDIR)/dummy | grep 'unused_function'
```

I tried to implement this check, but it turns out `$(TMPDIR)/dummy` is not accessible when run in the Makefile. When I ran the same command locally, it works like a charm. I'm under the impression there's a race condition within the Makefile. 

I `git grep`'d to find if a similar case existed in the code base, but I didn't manage to find any. Then, here's another proposal, closer to what already exist in the current Makefile.